subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -208,7 +208,18 @@ void bt_ots_dir_list_init(struct bt_ots_dir_list **dir_list, void *obj_manager)
 {
 	struct bt_gatt_ots_object *dir_list_obj;
 	int err;
+
+	__ASSERT(strlen(CONFIG_BT_OTS_DIR_LIST_OBJ_NAME) <= CONFIG_BT_OTS_OBJ_MAX_NAME_LEN,
+		 "BT_OTS_DIR_LIST_OBJ_NAME shall be less than %u octets",
+		 CONFIG_BT_OTS_OBJ_MAX_NAME_LEN);
+
+#if defined(CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT)
+	static char dir_list_obj_name[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
+
+	strcpy(dir_list_obj_name, CONFIG_BT_OTS_DIR_LIST_OBJ_NAME);
+#else
 	static char *dir_list_obj_name = CONFIG_BT_OTS_DIR_LIST_OBJ_NAME;
+#endif
```

Comment:
Even if we allow the name to be writable, it's up to each object to have that property. Do we really want the directory listing object to be writable?

Reply:
I don't think we have per object granularity on the object name characteristic's properties. It is my understanding that the object name characteristic is configured with a writable option and that applies to all object names as the objects are selected via the OLCP.

Reply:
You are absolutely correct :)